drivers/timer/arm_arch_timer.c

```diff
@@ -10,10 +10,9 @@
 #include <spinlock.h>
 #include <arch/cpu.h>
 
-#define CYC_PER_TICK	((uint32_t)((uint64_t)sys_clock_hw_cycles_per_sec() \
-			       / (uint64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC))
-
-#define MAX_TICKS	((0xffffffffu - CYC_PER_TICK) / CYC_PER_TICK)
+#define CYC_PER_TICK	((uint64_t)sys_clock_hw_cycles_per_sec() \
+			/ (uint64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC)
```

Comment:
FWIW: sys_clock_hw_cycles_per_sec() is a C API that is identical to CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC on all platforms except the hpet driver on x86.  There's not a lot of value to using it here, it probably got cut-and-pasted from another timer driver, which was itself probably modified via a script when the wrapper API was introduced.

Really hardware frequency values like this want to be coming from devicetree anyway.

Reply:
Noted. 
Typically we would read the counter frequency from Cortex-A core register cntfrq which is expected to be programmed by EL3 stage. ie. It could be pre-boot firmware or arch code in zephyr it self.  Maybe another PR someday.